### ISSUE TITLE
Extension controller removal: Ensure tooltips don't break for users that have opted-out of codemirror

### ISSUE THREAD

**Issue by philipp-spiess** [sourcegraph/sourcegraph#46240]
c.f. https://github.com/sourcegraph/sourcegraph/pull/45926#issuecomment-1372547613 and the attached Slack channel.

With the removal of the extension controller, some feature flag combinations will no longer work as expected (e.g. html-blob view will no longer have working tooltips).

We need to make sure that defaults are properly migrated or fix this in another way

/cc @felixfbecker

**Comment by philipp-spiess** [comment]
Okay so I decided to revert #45926 for now (c.f. #46251). I'll bring it back once we have ruled out the issues. Here are some thoughts:

1. **We need to enable CodeMirror and token-based navigation by default.** I mistakenly thought that the [PR](https://github.com/sourcegraph/sourcegraph/pull/45514) for that already got merged, but it didn't. Without this change, the extension controller change breaks `main`. I did some research and the PR I was referring to was only about enabling these features on dotcom. These settings are now out on S2 and k8s as well but we need to make sure this also ships as the default in the jan release. **We need to flip the defaults before relanding #45926 to avoid breaking `main`**.
2. Users that have `enableCodeMirrorFileView` manually set to `false` or `codeNavigation` manually set to something other than `"selection-driven"` will experience issues even if we change the defaults. This is what caused confusion [in Slack](https://sourcegraph.slack.com/archives/CHXHX7XAS/p1672910607988879). There are a couple of options I see here:
  - We do nothing. I don't expect the regular user to tamper with these feature flags much, mostly to enable the feature but not to force-disable something. _This is my recommended approach_ obviously :P.
  -  We completely remove the feature flag, so if someone has ` "enableCodeMirrorFileView": false` set, it won't do anything. This is risky since we will not have an escape hatch if we do realize there's something wrong with code mirror. Counterpoint: If this is the case, we should really work on a patch release ASAP instead of having customers tinker around with these settings since this will affect everyone (blob view is our most loaded page).
  - We add a new feature flag. This will avoid clashes with old `enableCodeMirrorFileView` overwrites. 
  - We work on a migration that resets the state of `enableCodeMirrorFileView` for everyone. IMO this is WAY out of scope for this issue.
